lib/pos_block_producer.go

```diff
@@ -182,13 +185,18 @@ func (pbp *PosBlockProducer) getBlockTransactions(
 		return nil, 0, errors.Wrapf(err, "Error copying UtxoView: ")
 	}
 	for _, txn := range feeTimeTxns {
+		// If we've exceeded the soft max block size, we exit.
+		if currentBlockSize > softMaxBlockSizeBytes {
+			break
```

Comment:
Add a comment here on why we `break` and don't just `continue`. TDLR, we want to allow at least one txn that moves the cumulative block size past the soft max, but don't want to add more txns beyond that.